clients/drcachesim/analysis_tool.h

```diff
@@ -129,6 +139,16 @@ template <typename RecordType> class analysis_tool_tmpl_t {
     {
         return initialize();
     }
+    /**
+     * Identifies the shard type for this analysis.  The return value indicates whether
+     * the tool supports this shard type, with failure (a non-empty string) being treated
+     * as a fatal error for the analysis.
+     */
+    virtual std::string
+    initialize_shard_type(shard_type_t shard_type)
```

Comment:
Right, it's a tradeoff -- but I would vote for this separate dedicated function.

Reply:
Or maybe a struct arg so that extending it to include new init args is easier?

Reply:
I don't see a change addressing this in 6aaf8a0891fb516169398b8272595d696d2a929e. Maybe it's still in your local workspace?

Reply:
Oh, missed this from Friday.  Will get back to this PR soon -- other work has preempted it.

Reply:
We had an offline discussion on this.

We had several design choices:

+ Dedicated `initialize_xxx` with dedicated error value showing whether this new thing is supported where tools can ignore and only have to override if they care about that thing; yet it can be confusing to have multiple init routines.

+ Future-proof `initialize_aux_info(const aux_info_t &info)` with `struct aux_info_t { shard_type_t shard_type; // Add new fields here w/o breaking backward compat. }` (plus maybe a `size` field for forward compat too).  This is separate from `initialize_stream` and so optional again whether override.

+ New `initialize_tool(const analysis_info_t &info)` which replaces `initialize_stream()` (we mark that as deprecated) and we put the stream and shard type as fields into the struct.  But now we (and users) have to go modify all their tools to use the new thing.  It is simpler for brand-new tools.

Given that there are trade-offs for each we're leaning toward just going with the dedicated init routine that's in the current PR.  We do not anticipate any future cases adding new things to init, since most are known to the stream and we can add to the stream interface.

Reply:
Added a comment to the header.